File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {

        int low=0; 
        int high= nums.length-1; 
        int mid=1; 

        while(low<=high){
             mid= low+(high-low)/2; 
            if(nums[mid]==target)
                return mid; 

            if(nums[mid]>target)
                high= mid-1; 
            
            else
                low=mid+1; 
        }
        return low;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, t):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        lo=0
        hi= len(nums)-1

        while lo<=hi: 
            mid = lo+(hi-lo)//2
            if t==nums[mid] : 
                return mid
            elif t<nums[mid]: 
                hi=mid-1 
            else: 
                lo = mid+1
        return lo 


